cmd/block-generator/generator/generate.go

```diff
@@ -450,33 +451,44 @@ func (g *generator) generatePaymentTxn(round uint64, intra uint64) (transactions
 	return g.generatePaymentTxnInternal(selection.(TxTypeID), round, intra)
 }
 
+const minBal = uint64(100000)
+
 func (g *generator) generatePaymentTxnInternal(selection TxTypeID, round uint64, intra uint64) (transactions.SignedTxn, transactions.ApplyData, error) {
 	defer g.recordData(track(selection))
 
+	// amounts
+	amount := uint64(1)
+	fee := uint64(1000)
+
+	// Select a receiver
 	var receiveIndex uint64
 	switch selection {
 	case paymentTx:
 		receiveIndex = rand.Uint64() % g.numAccounts
 	case paymentAcctCreateTx:
+		// give new accounts extra algos for sending other transactions (optin/optout)
+		amount = minBal * 100
 		g.balances = append(g.balances, 0)
-		g.numAccounts++
-		receiveIndex = g.numAccounts - 1
+		receiveIndex = g.numAccounts
+		// increment at the end in case it needs to be referenced later.
+		defer func() {
+			g.numAccounts++
+		}()
 	}
+	total := amount + fee
 
-	// Always send from a genesis account.
-	sendIndex := g.numPayments % g.config.NumGenesisAccounts
+	// Select a sender from genesis account
+	sendIndex := (g.numPayments + g.paymentOffset) % g.config.NumGenesisAccounts
```

Comment:
why is `paymentOffset` used here?

Reply:
I was thinking we could have gotten into a situation where the genesis accounts were exhausted and we were now sending payments from non-genesis accounts. In that scenario keeping track of an offset would avoid checking the same non-genesis accounts over and over again.

After the other changes, I think it would take a very long to hit this condition, and by then there are probably many other problems with the generator that I'm not yet aware of.